packages/expo-updates/android/src/main/java/expo/modules/updates/reloadscreen/ReloadScreenManager.kt

```diff
@@ -0,0 +1,105 @@
+package expo.modules.updates.reloadscreen
+
+import android.app.Activity
+import android.view.ViewGroup
+import com.facebook.react.bridge.ReactMarker
+import com.facebook.react.bridge.ReactMarkerConstants
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import java.lang.ref.WeakReference
+
+object ReloadScreenManager {
```

Comment:
any chance we can prevent the singleton usage and maybe move the ownership into EnabledAppController?

Reply:
singleton is hard to mock and write unit tests. the ownership/lifecycle is not quite clear. it's just my preference though. if that's not worth let's forget this.

Reply:
I understand, ideally we should avoid singletons. I was just hesitant to change existing api to accommodate this but if you feel that's better, we can do it

Reply:
well let's keep it as-is. we can improve that in the future. (or even introduced a DI framework 🙈)

Reply:
A second vote from me for not using a singleton if possible. Should be doable in a follow-up refactor though as well.